### PR TITLE
Tolerate timestamps w/ truncated nanos.

### LINKED ISSUE #1783: Timestamp nanos parsing broken for '2016-05-09T18:49:00.655319Z'
```
ERROR: test_log_text (logging_.TestLogging)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/system_tests/logging_.py", line 79, in test_log_text
    entries, _ = logger.list_entries()
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/logging/logger.py", line 317, in list_entries
    page_size=page_size, page_token=page_token)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/logging/client.py", line 141, in list_entries
    for resource in resp.get('entries', ())]
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/logging/client.py", line 141, in <listcomp>
    for resource in resp.get('entries', ())]
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/logging/client.py", line 79, in _entry_from_resource
    return TextEntry.from_api_repr(resource, self, loggers)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/logging/entries.py", line 92, in from_api_repr
    timestamp = _rfc3339_nanos_to_datetime(timestamp)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/_helpers.py", line 344, in _rfc3339_nanos_to_datetime
    dt_str, _RFC3339_NANOS.pattern))
ValueError: Timestamp: '2016-05-09T18:49:00.655319Z', does not match pattern: '\n    (?P<no_fraction>\n        \\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}  # YYYY-MM-DDTHH:MM:SS\n    )\n    \\.                                       # decimal point\n    (?P<nanos>\\d{9})                         # nanoseconds\n    Z                                        # Zulu\n'
```

From https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/128927443

I haven't traced down the codepath just yet.

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1783.

### CODE REVIEW

File: gcloud/test__helpers.py [hunk 1]
```diff
@@ -483,20 +483,34 @@ def test_w_bogus_zone(self):
...
         minute = 44
         seconds = 32
-        micros = 123456
-
-        dt_str = '%d-%02d-%02dT%02d:%02d:%02d.%06dZ' % (
-            year, month, day, hour, minute, seconds, micros)
-        with self.assertRaises(ValueError):
-            self._callFUT(dt_str)
+        truncateds_and_micros = [
+            ('12345678', 123456),
```
Comment (dhermes) [inline gcloud/test__helpers.py]: Isn't this a rounding error?

Comment (tseaver) [inline gcloud/test__helpers.py]: We've deliberately been doing integer (floor) division to this point.

Comment (dhermes) [inline gcloud/test__helpers.py]: What does the standard library do? (Checking now.)

Comment (dhermes) [inline gcloud/test__helpers.py]:
```python
>>> import datetime
>>> datetime.datetime(2016, 5, 10, 18, 0, 24, 203380.999)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: integer argument expected, got float
>>> import calendar
>>> calendar.timegm((2016, 5, 10, 18, 0, 24, 203380))
1462903224
>>> calendar.timegm((2016, 5, 10, 18, 0, 24, 203380.999))
1462903224
```

Comment (dhermes) [inline gcloud/test__helpers.py]: I guess that isn't helpful since

```python
>>> calendar.timegm((2016, 5, 10, 18, 0, 24, 0))
1462903224
```

**Comment by rimey** [comment]
I don't have the bandwidth right now to follow this properly, but have you guys looked at the [proto3 JSON mapping spec for Timestamp](https://developers.google.com/protocol-buffers/docs/proto3#json) and the [`Timestamp` class in `google.protobuf`](https://github.com/google/protobuf/blob/master/python/google/protobuf/internal/well_known_types.py#L95). Could we use that library's support for JSON serialization and deserialization of the well-known type `Timestamp`?

**Comment by dhermes** [comment]
Possibly, the phrase

> Uses RFC 3339, where generated output will always be Z-normalized and uses 3, 6 or 9 fractional digits.

may actually shed some light on what we've seen empirically.

**Comment by tseaver** [comment]
@dhermes Seems like you didn't find a counter-example to the "truncate/floor to micros" bit.

**Comment by dhermes** [comment]
Yeah LGTM

**Pull request merged**